pkg/cli/cli.go

```diff
@@ -1321,12 +1320,12 @@ func runConnectionWebSocketAgentCommand(flags *ConnectFlags, c *command, args []
 		pos:    0,
 		buf:    nil,
 	}
-	defer wsWrapper.Close()
 	go func() {
 		io.Copy(wsWrapper, tcpConn)
 		wsWrapper.Close()
 	}()
 	io.Copy(tcpConn, wsWrapper)
+	tcpConn.Close()
```

Comment:
This one should stay as it was, with the `defer` in line 1275. It has the same effect of closing at the end of the function, but with the added benefit of closing the socket even with an early return in case of errors.

I wrongly suggested before that this one needed fixing because I had the order of parameters to `io.Copy` inverted. Sorry about that.